### PR TITLE
Adding ERSEM workflow docs

### LINKED ISSUE #62: Add a description of workflow for feature development and merging to the main code
Add a brief description of workflow for developing new features in the model and submitting them for inclusion in the main code, something like "Getting Started with ERSEM development".

### PR DESCRIPTION
**Pull request by wathen**
Fixes #62 

Adds documentation to describe the workflow to add new developments to the ERSEM code base. 

The built docs can be found [here](https://ersem--69.org.readthedocs.build/en/69/)

### CODE REVIEW

File: docs/source/workflow.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+2. Create a `branch <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository>`_
+   from the `public ERSEM <https://github.com/pmlmodelling/ersem>`_ repo. The naming
+   convention of the branch is as follows `XX-[description]` where `XX` is the issue ID (this can be found
+   after creation of the issue) and `[description]` is a one or two work description to identify the branch
+   with the corresponding issue.
+3. Commit all changes outlined in the issue to this branch and push branch to the ERSEM repo.
+4. Create `pull request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_
+   to merge the branch you have made the changes on into the `master` branch. In the creation of the pull request you
+   will need to assign specific people to review the changes you have made before it can be merged into the master branch.
+5. Once the updates have been approved by the review/reviewers they will merge the development into the master branch.
```
Comment (glessin) [inline docs/source/workflow.rst:76]: as above, "reviewer/reviewers", or "reviewer(s)" :)

File: docs/source/workflow.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+6. When the public master branch is updated you will need to talk to the ERSEM repo manager to merge those changes back
+   to the private repo.
+
+Notes for reviewers
+-------------------
+
+When `reviewing pull requests <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews>`_ there are a few things to check:
+
+1. Ensure that the new changes do not adversely effect ERSEM. **Note** some modifications will change the results
```
Comment (glessin) [inline docs/source/workflow.rst:85]: "adversely affect"
"reviewers' responsibility"

**Review by glessin — CHANGES_REQUESTED** [review]
All looks good to me, short and clear instructions. Only a couple of mostly grammar/style suggestions from me.

File: docs/source/workflow.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+`milestone <https://github.com/pmlmodelling/ersem/milestones>`_ which you can then assign multiple
+issues too.
+
+Private developments
+--------------------
+
+The following come under private developments:
+
+1. Private project developments - these developments require significant scientific changes to the model,
+   and thus, are made privately.
```
[the remark this answers is not in the document]
Comment (yutiPML) [reply docs/source/workflow.rst]: In the compiled version, I think "only" is in the wrong place? the sentence does not read ok. Maybe a better way is:
"private developments comes only for specific projects that require significant scientific...."

File: docs/source/workflow.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+   to merge the branch you have made the changes on into the `master` branch. In the creation of the pull request you
+   will need to assign specific people to review the changes you have made before it can be merged into the master branch.
+5. Once the updates have been approved by the review/reviewers they will merge the development into the master branch.
+6. When the private master branch is updated you will need to talk to the ERSEM repo manager to merge those changes back
+   to the public repo.
+
+Public developments
+-------------------
+
+The following come under public developments:
```
Comment (yutiPML) [inline docs/source/workflow.rst:51]: It took me a while to understand the meaning of this sentence. Once I got it, it was clear, but maybe something like that would be clearer.
"the following cases are considered public developments, and they will need to follow a different workflow (see below):"

File: docs/source/workflow.rst (hunk 2 above)
Comment (yutiPML) [inline docs/source/workflow.rst]: we might want to "define" adversely effect and the threshold level for changes to go into ERSEM.
With a model so interconnected, it is likely that changing one aspect will have consequences somewhere else: when the reviewer should highlight these? any non-computational difference should be highlighted (unless the author of the change already discussed those in the pull request) or only when these create big issues (e.g. unrealistic simulations)?
Furthermore, the adverse effect might only appear in certain condition: is there a reference set-up that reviewers need to run? or is the author of the change supposed to give it?

File: docs/source/workflow.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+Notes for reviewers
+-------------------
+
+When `reviewing pull requests <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews>`_ there are a few things to check:
+
+1. Ensure that the new changes do not adversely effect ERSEM. **Note** some modifications will change the results
+   produced by the model, however, it is the author of the changes and reviewers responsibility to decide
+   whether these changes go into ERSEM. If unsure, please contact the ERSEM repository manager.
+2. Ensure that the changes made to the code correctly address the issue created.
```
Comment (yutiPML) [inline docs/source/workflow.rst:87]: similarly to the above, is the set-up provided by the author of the change?

File: docs/source/workflow.rst [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+3. For the public ERSEM, make sure that the tests continue to pass.
```
Comment (yutiPML) [inline docs/source/workflow.rst:88]: a link to the "how to check if ERSEM is passing the tests" would be helpful

File: docs/source/workflow.rst [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+For larger project developments that require multiple issues, it is often preferable to create a
+`milestone <https://github.com/pmlmodelling/ersem/milestones>`_ which you can then assign multiple
+issues too.
+
+Private developments
+--------------------
+
+The only private developments would come from 
+specific projects - these require significant scientific changes to the model,
+and thus, are isolated from the public version of the code.
```
Comment (yutiPML) [inline docs/source/workflow.rst:26]: I initially struggled to understand the sentence, maybe something like would be clearer
"Private developments are those generated within new and ongoing projects that requires significant scientific changes to the code that will be initially isolated from the public version of the code"

File: docs/source/workflow.rst [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+   to merge the branch you have made the changes on into the `master` branch. In the creation of the pull request you
+   will need to assign specific people to review the changes you have made before it can be merged into the master branch.
+5. Once the updates have been approved by the reviewer/reviewers they will merge the development into the master branch.
+6. When the private master branch is updated you will need to talk to the ERSEM repo manager to merge those changes back
+   to the public repo.
+
+Public developments
+-------------------
+
+The following come under public developments:
```
Comment (yutiPML) [inline docs/source/workflow.rst:51]: similar to the above comment, maybe something like
"the following changes are considered public developments and would need to follow a slightly different workflow (see below)"

File: docs/source/workflow.rst [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+   to the private repo.
+
+Notes for reviewers
+-------------------
+
+When `reviewing pull requests <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews>`_ there are a few things to check:
+
+1. Ensure that the new changes do not adversely affect ERSEM. **Note** some modifications will change the results
+   produced by the model, however, it is the author of the changes and reviewer's responsibility to decide
+   whether these changes go into ERSEM. If unsure, please contact the ERSEM repository manager.
```
Comment (yutiPML) [inline docs/source/workflow.rst:86]: It is not clear what is an "adverse effect": result in a variable the id not directly connected to the change being numerically different? or an unreasonable behaviour? and what is the threshold of unreasonable?

Comment (yutiPML) [reply docs/source/workflow.rst:86]: Furthermore, it is not clear how the effect should be checked.
with which set-up? who's responsible to provide the set-up?

File: docs/source/workflow.rst [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+2. Ensure that the changes made to the code correctly address the issue created.
```
Comment (yutiPML) [inline docs/source/workflow.rst:87]: similarly to the above, is the set-up provided by the author of the change?

File: docs/source/workflow.rst [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+3. For the public ERSEM, make sure that the tests continue to pass.
```
Comment (yutiPML) [inline docs/source/workflow.rst:88]: a link to the "how to check if ERSEM is passing the tests" would be helpful

**Review by yutiPML — CHANGES_REQUESTED** [review]

File: docs/source/workflow.rst (hunk 2 above)
Comment (wathen) [reply docs/source/workflow.rst]: Unfortunately it is hard to define a threshold level for changes, maybe re-wording to ensure that people will start a conversation about the changes with other members of the group would be best?

File: docs/source/workflow.rst (hunk 8 above)
Comment (wathen) [reply docs/source/workflow.rst:51]: Done

File: docs/source/workflow.rst (hunk 7 above)
Comment (wathen) [reply docs/source/workflow.rst:26]: Done

**Pull request merged**